### DISCUSSION TITLE
ch 07 퀴즈

### DISCUSSION THREAD

**Discussion by Sleepingoff** [OurStudyLog/operatingsystem#10] [Q&A] [answered]
플래시 메모리의 다음 그림을 보고 어떤 일이 일어나는지 각 단계에 대해 설명하시오.
![image](https://github.com/OurStudyLog/operatingsystem/assets/111721483/136eb393-559a-4425-88b2-27b13e4cec0c)

**Comment by ImSeon01** [comment]
플래시 메모리는 쓰기와 삭제 단위가 다르다.
쓰기는 페이지 단위로 하지만
삭제는 블록 단위로 이루어진다.

1번 단계에서는 페이지 C를 생성하였다.

2번 단계는 A라는 페이지를 인벨리트 상태로 만들고 A'라는 페이지를 새로 생성하였다. 

3번 단계에서 가비지 컬렉터가 유효하지 않은 페이지 A를 삭제 하였다.

**Comment by uzjaee** [comment, marked as answer]
1단계 
블록 내에 valid 상태 페이지를 하나 추가합니다. ( 읽고 쓰는 것은 페이지 단위) 
2단계
A를 Invalid 상태로 만들고 A' 를 복사합니다. 
3단계 
전체 블록을 복사하여 다른 블록을 추가 하되 하나의 페이지는 가져가지 않습니다 . 
( 블록 단위로 삭제 해야하기 때문)

**Comment by atoonatoo** [comment]
1번의 경우 valid상태로 
2번의 경우 invalid 상태
3번의
